bmaptools/TransRead.py

```diff
@@ -225,140 +263,37 @@ def _open_compressed_file(self):
-                return True
-            return False
-
-        def is_lzop(name):
-            """Returns 'True' if file 'name' is compressed with 'lzop'."""
-            if name.endswith('.lzo') and not name.endswith('.tar.lzo'):
-                return True
-            return False
-
-        def is_lz4(name):
-            """Returns 'True' if file 'name' is compressed with 'lz4'."""
-            if name.endswith('.lz4') and not name.endswith('.tar.lz4'):
-                return True
-            return False
-
-        def is_tar_gz(name):
-            """
-            Returns 'True' if file 'name' is a tar archive compressed with
-            'gzip'.
-            """
-
-            if name.endswith('.tar.gz') or name.endswith('.tgz'):
-                return True
-            return False
-
-        def is_tar_bz2(name):
-            """
-            Returns 'True' if file 'name' is a tar archive compressed with
-            'bzip2'.
-            """
-
-            if name.endswith('.tar.bz2') or name.endswith('.tbz') or \
-               name.endswith('.tbz2') or name.endswith('.tb2'):
-                return True
-            return False
-
-        def is_tar_xz(name):
-            """
-            Returns 'True' if file 'name' is a tar archive compressed with 'xz'.
-            """
-
-            if name.endswith('.tar.xz') or name.endswith('.txz'):
-                return True
-            return False
-
-        def is_tar_lzo(name):
-            """
-            Returns 'True' if file 'name' is a tar archive compressed with
-            'lzop'.
-            """
-
-            if name.endswith('.tar.lzo') or name.endswith('.tzo'):
-                return True
-            return False
-
-        def is_tar_lz4(name):
-            """
-            Returns 'True' if file 'name' is a tar archive compressed with
-            'lz4'.
-            """
-
-            if name.endswith('.tar.lz4') or name.endswith('.tlz4'):
-                return True
-            return False
+        def _get_archive_extension(name):
+            for extension in NESTED_EXTENSION_ALIASES:
+                if name.endswith(extension):
+                    return NESTED_EXTENSION_ALIASES[extension]
+            for extension in EXTENSION_ALIASES:
+                if name.endswith(extension):
+                    return EXTENSION_ALIASES[extension]
+            for extension in NESTED_DECOMPRESSER_EXTENSIONS:
+                if name.endswith(extension):
+                    return extension
+            for extension in DECOMPRESSER_EXTENSIONS:
+                if name.endswith(extension):
+                    return extension
+            return None
 
         archiver = None
-        if is_tar_gz(self.name) or is_gzip(self.name):
-            self.compression_type = 'gzip'
-            if BmapHelpers.program_is_available("pigz"):
+        archive_extension = _get_archive_extension(self.name)
+        if archive_extension is not None:
```

Comment:
May be worth restructuring like this:

if archive_extension is None:
   blah
   return

blah blah - large block of code

Just for readability and simplicity. But this is just a nitpick, no strong opinion.